packages/cardpay-sdk/sdk/token-bridge-foreign-side.ts

```diff
@@ -85,6 +85,22 @@ export default class TokenBridgeForeignSide implements ITokenBridgeForeignSide {
     let foreignBridge = await getAddress('foreignBridge', this.layer1Web3);
     let nextNonce = await this.getNextNonce(from);
 
+    let withinLimit = await this.withinLimit(tokenAddress, amount);
+    if (!withinLimit) {
+      let maxPerTx = await this.maxPerTx(tokenAddress);
+      let minPerTx = await this.minPerTx(tokenAddress);
+      let dailyLimit = await this.dailyLimit(tokenAddress);
+      let totalSpentPerDay = await this.totalSpentPerDay(tokenAddress);
+      throw new Error(`
+        unlock action NOT within configured limits
+
+          Minimum Amount Per Tx: ${fromWei(minPerTx)}
+          Maximum Amount Per Tx: ${fromWei(maxPerTx)}
+          Total Amount Today: ${fromWei(totalSpentPerDay)}
+          Daily Amount Limit: ${fromWei(dailyLimit)}
+
+      `);
+    }
```

Comment:
For this and for `relayTokens`, do we want to include the deposited amount in the error message?

Reply:
The types for minPerTx, maxPerTx, dailyLimit, and totalSpentPerDay may not be correct - they don't seem to be BNs, leading to a problem here: https://github.com/cardstack/cardstack/pull/3134#partial-pull-merging . Should they be strings or BNs?

Reply:
Thanks for the suggestions! I will make the corresponding change. 
- add deposited amount to `unlock` and `relay` 
- covert those minPerTx, maxPerTx, dailyLimit, and totalSpentPerDay  to BNs

Reply:
EDIT: It looks like the call() is returning a string. I will adjust the types. And ur conversion occured it other PR will stay as it is. Sorry bout that

Reply:
@Aierie I made the BN change. And I removed the conversion u did to getDepositLimits() previously 
as stated here https://github.com/cardstack/cardstack/pull/3134#discussion_r935719355